### PR TITLE
Use the new code generator API for LLVM backend.

### PR DESCRIPTION
**Pull request by losfair**

### CODE REVIEW

**Comment by bjfish** [comment]
bors try

**Comment by bjfish** [comment]
bors try

**Comment by bjfish** [comment]
bors try

**Comment by bjfish** [comment]
bors try

**Comment by bjfish** [comment]
bors try

**Comment by bjfish** [comment]
bors try

**Comment by bjfish** [comment]
bors cancel

**Comment by bjfish** [comment]
bors try

File: lib/llvm-backend/src/code.rs [hunk 1]
```diff
@@ -2139,324 +2402,245 @@ fn parse_function(
...
+            [] => {
+                self.builder.as_ref().unwrap().build_return(None);
+            }
+            [one_value] => {
+                self.builder.as_ref().unwrap().build_return(Some(one_value));
+            }
+            _ => {
+                // let struct_ty = llvm_sig.get_return_type().as_struct_type();
+                // let ret_struct = struct_ty.const_zero();
+                unimplemented!("multi-value returns not yet implemented")
```
Comment (xmclark) [inline lib/llvm-backend/src/code.rs]: need these comments still?

File: lib/llvm-backend/src/code.rs [hunk 2]
```diff
@@ -2139,324 +2402,245 @@ fn parse_function(
...
+        // pass_manager.add_instruction_combining_pass();
+        pass_manager.add_aggressive_inst_combiner_pass();
+        pass_manager.add_merged_load_store_motion_pass();
+        // pass_manager.add_sccp_pass();
+        // pass_manager.add_gvn_pass();
+        pass_manager.add_new_gvn_pass();
+        pass_manager.add_aggressive_dce_pass();
+        pass_manager.run_on_module(&self.module);
+
+        // self.module.print_to_stderr();
```
Comment (xmclark) [inline lib/llvm-backend/src/code.rs]: do these comments need to be here?

Comment (bjfish) [reply lib/llvm-backend/src/code.rs]: I've cleaned up these comments.

File: lib/llvm-backend/src/code.rs (hunk 1 above)
Comment (bjfish) [reply lib/llvm-backend/src/code.rs]: I've removed these comments, thanks.

**Comment by bjfish** [comment]
bors try

**Pull request merged**